test/opaque_closure.jl

```diff
@@ -153,3 +153,30 @@ mk_va_opaque() = @opaque (x...)->x
 
 # OpaqueClosure show method
 @test repr(@opaque x->1) == "(::Any)::Any->◌"
+
+# Opaque closure in CodeInfo returned from generated functions
+function mk_ocg(args...)
+    ci = @code_lowered const_int()
+    cig = Meta.lower(@__MODULE__, Expr(:new_opaque_closure, Tuple{}, false, Any, Any,
+        Expr(:opaque_closure_method, 0, lno, ci))).args[1]
+    cig.slotnames = Symbol[Symbol("#self#")]
+    cig.slottypes = Any[Any]
+    cig.slotflags = UInt8[0x00]
+    cig
+end
+
+@eval function oc_trivial_generated()
+    $(Expr(:meta, :generated_only))
+    $(Expr(:meta,
+            :generated,
+            Expr(:new,
```

Comment:
Could this be written using `@generated` to depend less on implementation details?

Reply:
I don't think `@generated` are allowed to return CodeInfo anymore.

Reply:
When did that happen? So can we remove the rest of the code supporting it?

Reply:
Generated functions still support it, you just can't get there with the `@generated` syntax. If you try it, the macro will wrap it in an AST for you. No idea if that was intentional, but I assume it happened when the generated function internals were changed.